packages/drawer/src/ux-drawer.ts

```diff
@@ -0,0 +1,302 @@
+      }
+      if (this.defaultConfig.overlayDismiss !== undefined) {
+        this.overlayDismiss = this.defaultConfig.overlayDismiss
+      }
+      if (this.defaultConfig.position !== undefined) {
+        this.position = this.defaultConfig.position
+      }
+      if (this.defaultConfig.keyboard !== undefined) {
+        this.keyboard = this.defaultConfig.keyboard
+      }
+      if (this.defaultConfig.theme !== undefined) {
+        this.theme = this.defaultConfig.theme
+      }
+    }
+  private bindingContext: any;
+  public bind(bindingContext: any) {
+    this.bindingContext = bindingContext;
+    this.themeChanged(this.theme);
+    this.setViewportType();
+    window.addEventListener('resize', this);
+    this.positionChanged();
+    this.modalBreakpointChanged();
+    this.hostChanged();
+    this.overlayDismissChanged();
+    this.keyboardChanged();
+  }
+
+  public positionChanged() {
+    if (!this.position && this.defaultConfig.position) {
+      this.position = this.defaultConfig.position;
+    }
+  }
+
+  public modalBreakpointChanged() {
+    if (typeof this.modalBreakpoint !== 'number' && this.defaultConfig.modalBreakpoint) {
+      this.modalBreakpoint = this.defaultConfig.modalBreakpoint;
+    }
+  }
+
+  public hostChanged() {
+    if (this.host === false || this.host === 'body' || this.host instanceof HTMLElement) {
+      return;
+    }
+    if (this.defaultConfig.host !== undefined) {
+      this.host = this.defaultConfig.host;
+      return;
+    }
+    if (this.host === '') {
+      this.host = 'body';
+    }
+  }
+
+  public overlayDismissChanged() {
+    if (!this.overlayDismiss && this.defaultConfig.overlayDismiss) {
+      this.overlayDismiss = this.defaultConfig.overlayDismiss;
+    }
+  }
+
+  public keyboardChanged() {
+    if (!this.keyboard && this.defaultConfig.keyboard) {
+      this.keyboard = this.defaultConfig.keyboard;
+    }
+  }
+
+  public attached() {
+    if (this.host) {
+      this.moveToHost();
+    }
+    this.show();
+  }
+
+  public detached() {
+    if (this.host) {
+      this.removeFromHost();
+    }
+  }
+
+  private show() {
+    if (this.showing && this.showed) {return;}
+    if(DOM.activeElement instanceof HTMLElement) {
+      this.lastActiveElement = DOM.activeElement;
+    }
+    this.showing = true;
+    this.modalService.addLayer(this, this.bindingContext);
+    this.setZindex();
+    this.taskQueue.queueTask(() => {
+      this.showed = true;
+    });
+  }
+
+  private async hide() {
+    if (this.hiding || !this.showed) {return;}
+    this.hiding = true;
+    const duration = this.getAnimationDuration();
+    this.showed = false;
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        this.modalService.removeLayer(this);
+        this.hiding = false;
+        if (this.lastActiveElement && typeof this.restoreFocus === 'function') {
+          this.restoreFocus(this.lastActiveElement);
+        }
+        resolve();
+      }, duration + 10);
```

Comment:
can we have a small comment to explain this `+ 10`?

Reply:
I placed this `+ 10` here as a margin. But to be honest it doesn't quite make sense. I will rather remove it.